sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -25,17 +25,26 @@ import org.apache.spark.sql.catalyst.util.TypeUtils
 import org.apache.spark.sql.types._
 import org.apache.spark.util.Utils
 
+trait PredResult
```

Comment:
Can we restrict doing this only for the expressions that have this behavior for the expressions that need this?

This is just needed for a null aware equals right?